fastavro/_schema.pyx

```diff
@@ -208,51 +230,32 @@ cdef _parse_schema(
 ):
     # union schemas
     if isinstance(schema, list):
-        parsed_schemas = []
-        for index, s in enumerate(schema):
-            if index == 0:
-                parsed_schemas.append(
-                    _parse_schema(
-                        s,
-                        namespace,
-                        expand,
-                        False,
-                        names,
-                        named_schemas,
-                        default,
-                        ignore_default_error,
-                        in_union=True,
-                    )
-                )
+        parsed_schemas = [
+            _parse_schema(
+                s,
+                namespace,
+                expand,
+                False,
+                names,
+                named_schemas,
+                NO_DEFAULT,
+                ignore_default_error,
+            )
+            for s in schema
+        ]
+        if default is not NO_DEFAULT:
+            for s in parsed_schemas:
+                if _default_matches_schema(default, s):
+                    break
             else:
-                parsed_schemas.append(
-                    _parse_schema(
-                        s,
-                        namespace,
-                        expand,
-                        False,
-                        names,
-                        named_schemas,
-                        NO_DEFAULT,
-                        ignore_default_error,
-                    )
-                )
+                _raise_default_value_error(default, schema, True, ignore_default_error)
```

Comment:
Can you change this to

```
_raise_default_value_error(default, schema, in_union=True, ignore_default_error=ignore_default_error)
```

So that it's more clear what the `True` is for

Reply:
Resolved by removing is_union entirely.